File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.cpp
class Solution {
public:
    string removeStars(string s) {
        
        vector<char> st; 
        for(auto e:s){
            if(e !='*'){
                st.push_back(e) ;
            }else {
                st.pop_back(); 
            }
            
        }
        string ans ="";
        for(int i=0 ; i < st.size(); i++){
            ans += st[i] ; 
        }
        return ans ; 
        
    }
};

File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.java
class Solution {
    public String removeStars(String s) {
        
        char[] ch = new char[s.length()] ; 
        int j = 0 ; 

        for(int i=0; i < s.length() ; i++){
            char c = s.charAt(i) ; 
            if( c == '*' ){
                j-- ; 
            } else {
                ch[j++] = c ; 
            }
        }

        StringBuilder ans = new StringBuilder() ; 
        for(int i =0 ; i < j ; i++ ){
            ans.append(ch[i]) ;
        }
        return ans.toString() ; 


    }
}